fenix/app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt

```diff
@@ -579,26 +580,37 @@ abstract class BaseBrowserFragment :
         downloadFeature.onDownloadStopped = { downloadState, _, downloadJobStatus ->
             // If the download is just paused, don't show any in-app notification
             if (shouldShowCompletedDownloadDialog(downloadState, downloadJobStatus)) {
-                saveDownloadDialogState(
-                    downloadState.sessionId,
-                    downloadState,
-                    downloadJobStatus,
-                )
-
-                val dynamicDownloadDialog = DynamicDownloadDialog(
-                    context = context,
-                    downloadState = downloadState,
-                    didFail = downloadJobStatus == DownloadState.Status.FAILED,
-                    tryAgain = downloadFeature::tryAgain,
-                    onCannotOpenFile = {
-                        showCannotOpenFileError(binding.dynamicSnackbarContainer, context, it)
-                    },
-                    binding = binding.viewDynamicDownloadDialog,
-                    toolbarHeight = toolbarHeight,
-                ) { sharedViewModel.downloadDialogState.remove(downloadState.sessionId) }
+                val onCannotOpenFile: (DownloadState) -> Unit = {
```

Comment:
Logic-wise this is fine, but the nested logic is harder to read as we keep adding more things.

Reply:
Great, I'll update the patch to decouple this part :)